20_beta_diversity.Rmd

```diff
@@ -135,32 +127,54 @@ p <- p + labs(x = paste("PCoA 1 (", round(100 * rel_eig[[1]], 1), "%", ")", sep
 p
 ```
 
-With additional tools from the ggplot2 package, ordination methods can be 
-compared to find similarities between them or select the most suitable one to
-visualize beta diversity in the light of the research question.
+A few combinations of beta diversity metrics and assay types are conventionally
+used. For instance, Bray-Curtis dissimilarity and Aitchison distance are often applied to the relative abundance and the clr assays, respectively. Besides
+**beta diversity metric** and **assay type**, the **PCoA algorithm** is also a variable that should be considered. Below, we show how the choice of these three
+factors can affect the resulting lower-dimensional data.
 
-```{r plot-mds-nmds-comparison, fig.cap = "Comparison of MDS and NMDS plots based on the Bray-Curtis or euclidean distances on the GlobalPattern dataset."}
+```{r mds-nmds-comparison, warning=FALSE, message=FALSE}
+# Run NMDS on relabundance assay with Bray-Curtis distances
+tse <- runNMDS(tse,
+               FUN = vegan::vegdist,
+               method = "bray",
+               assay.type = "relabundance",
+               name = "NMDS_bray")
+
+# Run MDS on clr assay with Aitchison distances
 tse <- runMDS(tse,
               FUN = vegan::vegdist,
-              name = "MDS_euclidean",
-              method = "euclidean",
-              assay.type = "counts")
+              method = "aitchison",
+              assay.type = "clr",
+              pseudocount = 1,
```

Comment:
Why here is pseudocount, probably mistake

Reply:
Both here and below, an error arises if pseudocount is 0:
```
Error: 'clr' cannot be used with non-positive data: use pseudocount > 5.19440001749332e-05
```

Reply:
The way I think of it is the pseudocount in `transformAssay(method = "clr")` ensures that the input matrix is positive, but doesn't guarantee that the output matrix is completely positive

Reply:
Ahh ok, aitchison dissimilarity is CLR + Euclidean distance. https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5695134/

I did not know that it is possible to calculate CLR in vegdist. Also I remembered incorrectly Aithchison dissimilarity.

--> so the method should be here Euclidean distance

Reply:
pseudocount doesn't make sense in the context of CLR. It is added in order to calculate CLR (before CLR) if the data has zeroes - but never on top of CLR (after CLR).

It is possible to calculate CLR and rCLR transformations with vegan::decostand and their corresponding (r)Aitchison distances in vegan::vegdist, after my last years PR in vegan: 
https://github.com/vegandevs/vegan/pulls?q=is%3Apr+author%3Aantagomir+

Zero imputation procedure for the rCLR / rAitchison is yet to be added.